### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by tomin11**
- used arrays for multiple sensor variables,
- added launch file, 
- changed ROS message type to array

### CODE REVIEW

File: src/vl53l0x_driver.cpp [hunk 1]
```diff
@@ -1,48 +1,43 @@
...
-VL53L0X_Dev_t *pSensorOne = &SensorOne;
-VL53L0X_Dev_t *pSensorTwo = &SensorTwo;
-VL53L0X_Dev_t *pSensorThree = &SensorThree;
-VL53L0X_Dev_t *pSensorFour = &SensorFour;
-
-VL53L0X_RangingMeasurementData_t    SensorOneRangingMeasurementData;
-VL53L0X_RangingMeasurementData_t    SensorTwoRangingMeasurementData;
-VL53L0X_RangingMeasurementData_t    SensorThreeRangingMeasurementData;
-VL53L0X_RangingMeasurementData_t    SensorFourRangingMeasurementData;
+#define I2C_BUS_PATH "/dev/i2c-10"
```
Comment (bhuvanchandradv) [inline src/vl53l0x_driver.cpp]: Add these as ros params and set in launch file..

File: src/vl53l0x_driver.cpp [hunk 2]
```diff
@@ -1,48 +1,43 @@
...
+int VL53L0X_ADDR[4];
+VL53L0X_Dev_t Sensors[NUM_SENSORS];
+VL53L0X_Dev_t *pSensors[NUM_SENSORS];
+VL53L0X_RangingMeasurementData_t SensorsRangingMeasurementData[NUM_SENSORS];
+
+void declare() {
+  for (int i = 0; i < 15; i++)
+    VL53L0X_XSHUT_MCP23xx_IO[i] = i;
+
+  int i, j;
```
Comment (bhuvanchandradv) [inline src/vl53l0x_driver.cpp]: Be consistent, use local variable in function scope or use variable def in loop

**Review by bhuvanchandradv — CHANGES_REQUESTED** [review]

File: src/vl53l0x_driver.cpp (hunk 2 above)
Comment (tomin11) [reply src/vl53l0x_driver.cpp]: okay

File: src/vl53l0x_driver.cpp (hunk 1 above)
Comment (tomin11) [reply src/vl53l0x_driver.cpp]: okay

File: src/vl53l0x_driver.cpp [hunk 3]
```diff
@@ -1,185 +1,149 @@
...
+vl53l0x_driver::vl53l0x sensor_msg_array[NUM_SENSORS];
+int i2c_bus_instance;
+std::string i2c_bus_path;
+
+void initialize() {
+  for (int i = 0; i < NUM_SENSORS; i++) {
+    VL53L0X_XSHUT_MCP23xx_IO[i] = i;
+    pSensors[i] = &Sensors[i];
+  }
+  for (int i = 0, j = 0x21; i < NUM_SENSORS; j++, i++)
```
Comment (bhuvanchandradv) [inline src/vl53l0x_driver.cpp]: Can we do something like this:
```suggestion
for (int i = 0; i < NUM_SENSORS; i++) {
    VL53L0X_XSHUT_MCP23xx_IO[i] = i;
    pSensors[i] = &Sensors[i];
    VL53L0X_ADDR[i] = (0x21 + i);
}

File: src/vl53l0x_driver.cpp [hunk 4]
```diff
@@ -1,185 +1,141 @@
...
+  }
+
+  GPIO_Setup();
+  Sensor_Setup();
+
+  // Step 4 :: Calibration Sensor and Get Sensor Value
+  for (int i = 0; i < NUM_SENSORS; i++) {
+    Sensor_Calibration(pSensors[i]);
+  }
+  while (ros::ok()) {
```
Comment (bhuvanchandradv) [inline src/vl53l0x_driver.cpp]: Also check if i2c device file is available before publishing:
e.g.:
```suggestion
check_device_connection(int _file_descriptor) {
    struct stat device_status;
    fstat(_file_descriptor, &device_status);
    return device_status.st_nlink;
}
  while (ros::ok() && check_device_connection(pSensors[0]->fd)) {
```

File: src/vl53l0x_driver.cpp [hunk 5]
```diff
@@ -1,185 +1,149 @@
+#include <cstring>
 #include <ros/ros.h>
-#include "std_msgs/Int16.h"
-#include <stdlib.h>
+#include <std_msgs/Int16.h>
```
Comment (bhuvanchandradv) [inline src/vl53l0x_driver.cpp]: drop it

File: src/vl53l0x_driver.cpp [hunk 6]
```diff
@@ -1,185 +1,149 @@
...
+}
 
 #include "vl53l0x_api.h"
-#include "vl53l0x_platform.h"
-
-#include "vl53l0x_driver/getSensorData.h"
 #include "vl53l0x_driver/vl53l0x.h"
- 
+#include "vl53l0x_platform.h"
 #define MCP_ADDRESS 0x20
```
Comment (bhuvanchandradv) [inline src/vl53l0x_driver.cpp:15]: keep one line space here

File: src/vl53l0x_driver.cpp [hunk 7]
```diff
@@ -1,185 +1,149 @@
...
 
-bool getSensorData(vl53l0x_driver::getSensorData::Request &req, vl53l0x_driver::getSensorData::Response &res) {
-    res.One = SensorOneRangingMeasurementData.RangeMilliMeter;
-    res.Two = SensorTwoRangingMeasurementData.RangeMilliMeter;
-    res.Three = SensorThreeRangingMeasurementData.RangeMilliMeter;
-    res.Four = SensorFourRangingMeasurementData.RangeMilliMeter;
-    return true;
-}
-
 void GPIO_Setup() {
```
Comment (bhuvanchandradv) [inline src/vl53l0x_driver.cpp]: error handling

File: src/vl53l0x_driver.cpp [hunk 8]
```diff
@@ -1,185 +1,149 @@
...
+  if (i2c_mcp23017 == NULL)
+    return;
+
+  for (int i = 0; i < NUM_SENSORS; i++) {
+    mcp_pinMode(i2c_mcp23017, VL53L0X_XSHUT_MCP23xx_IO[i], OUTPUT);
+    mcp_digitalWrite(i2c_mcp23017, VL53L0X_XSHUT_MCP23xx_IO[i], LOW);
+  }
 }
 
 void Sensor_Setup() {
```
Comment (bhuvanchandradv) [inline src/vl53l0x_driver.cpp]: error handling

File: src/vl53l0x_driver.cpp [hunk 9]
```diff
@@ -1,185 +1,149 @@
...
+
+  ros::Publisher sensor_pub_array[NUM_SENSORS];
+  std::string name = "sensor_data_";
+  for (int i = 0; i < NUM_SENSORS; i++) {
+    std::string result = name + std::to_string(i + 1);
+    sensor_pub_array[i] = nh.advertise<vl53l0x_driver::vl53l0x>(result, 1000);
+  }
+
+  GPIO_Setup();
+  Sensor_Setup();
```
Comment (bhuvanchandradv) [inline src/vl53l0x_driver.cpp]: error handling

File: src/vl53l0x_driver.cpp [hunk 10]
```diff
@@ -1,185 +1,149 @@
...
+  ros::NodeHandle nh;
+  nh.getParam("/measure_proximity_node/i2c_bus_instance", i2c_bus_instance);
+  ROS_INFO("i2c_bus_instance: %d", i2c_bus_instance);
+  i2c_bus_path = "/dev/i2c-" + std::to_string(i2c_bus_instance);
+
+  ros::Publisher sensor_pub_array[NUM_SENSORS];
+  std::string name = "sensor_data_";
+  for (int i = 0; i < NUM_SENSORS; i++) {
+    std::string result = name + std::to_string(i + 1);
+    sensor_pub_array[i] = nh.advertise<vl53l0x_driver::vl53l0x>(result, 1000);
```
Comment (bhuvanchandradv) [inline src/vl53l0x_driver.cpp]: reduce the queue size

Comment (bhuvanchandradv) [reply src/vl53l0x_driver.cpp]: also can you move this to the for loop you used for calibration below.

**Review by bhuvanchandradv — APPROVED** [review]

File: src/vl53l0x_driver.cpp (hunk 3 above)
Comment (tomin11) [reply src/vl53l0x_driver.cpp]: yes

File: src/vl53l0x_driver.cpp (hunk 4 above)
Comment (tomin11) [reply src/vl53l0x_driver.cpp]: ok

File: src/vl53l0x_driver.cpp (hunk 6 above)
Comment (tomin11) [reply src/vl53l0x_driver.cpp:15]: ok

File: src/vl53l0x_driver.cpp (hunk 10 above)
Comment (tomin11) [reply src/vl53l0x_driver.cpp]: ok

File: src/vl53l0x_driver.cpp (hunk 9 above)
Comment (tomin11) [reply src/vl53l0x_driver.cpp]: ok

File: src/vl53l0x_driver.cpp (hunk 7 above)
Comment (tomin11) [reply src/vl53l0x_driver.cpp]: ok

File: src/vl53l0x_driver.cpp (hunk 8 above)
Comment (tomin11) [reply src/vl53l0x_driver.cpp]: ok

File: src/vl53l0x_driver.cpp (hunk 5 above)
Comment (tomin11) [reply src/vl53l0x_driver.cpp]: ok

**Pull request merged**